ddl/foreign_key.go

```diff
@@ -281,6 +281,68 @@ func checkTableForeignKey(referTblInfo, tblInfo *model.TableInfo, fkInfo *model.
 	return nil
 }
 
+func checkModifyColumnWithForeignKeyConstraint(is infoschema.InfoSchema, dbName string, tbInfo *model.TableInfo, originalCol, newCol *model.ColumnInfo) error {
+	if newCol.GetType() == originalCol.GetType() && newCol.GetFlen() == originalCol.GetFlen() && newCol.GetDecimal() == originalCol.GetDecimal() {
+		return nil
+	}
+	// WARN: is maybe nil.
+	if is == nil {
+		return nil
+	}
+	for _, fkInfo := range tbInfo.ForeignKeys {
+		for i, col := range fkInfo.Cols {
+			if col.L == originalCol.Name.L {
+				if !is.TableExists(fkInfo.RefSchema, fkInfo.RefTable) {
+					continue
+				}
+				referTable, err := is.TableByName(fkInfo.RefSchema, fkInfo.RefTable)
+				if err != nil {
+					return err
+				}
+				referCol := model.FindColumnInfo(referTable.Meta().Columns, fkInfo.RefCols[i].L)
+				if referCol == nil {
+					continue
+				}
+				if newCol.GetType() != referCol.GetType() {
+					return dbterror.ErrFKIncompatibleColumns.GenWithStackByArgs(originalCol.Name, fkInfo.RefCols[i], fkInfo.Name)
+				}
+				if newCol.GetFlen() < referCol.GetFlen() || newCol.GetFlen() < originalCol.GetFlen() ||
+					(newCol.GetType() == mysql.TypeNewDecimal && (newCol.GetFlen() != originalCol.GetFlen() || newCol.GetDecimal() != originalCol.GetDecimal())) {
+					return dbterror.ErrForeignKeyColumnCannotChange.GenWithStackByArgs(originalCol.Name, fkInfo.Name)
+				}
```

Comment:
Some codes are the same with the below codes. Can we reuse it ?

Reply:
It's not exactly the same, then hard to reuse it.